### PR TITLE
Fix completion issues

### LINKED ISSUE #20202: Chain completion suggestions are not working after the annotation access expressions
**Description:**
Chain completion suggestions are not working after the annotation access expressions.

**Steps to reproduce:**
```ballerina
type Annot record {
    int foo
};

public annotation Annot v1 on type;
@v1 {
    foo: 1
}
public type T1 record {
    string name;
};

T1 a = { name: "John" };

function testAnnotationAccessExpression() {
	typedesc<T1> t = typeof a;
        Annot? annotationVal = t.@v1.[cursor]
}
```

**Affected Versions:**
1.0.5

### LINKED ISSUE #28788: [LS] text/completion' failed with a NoSuchElementException
Sample program to reproduce 

```ballerina
import ballerina/http;

const BALLERINA_ORG = "ballerina-platform";
const BALLERINA_LANG_REPO = "ballerina-lang";

public function main() returns error? {
    check getLables(BALLERINA_ORG, BALLERINA_LANG_REPO);
}

function getLables(string org, string repo) returns error? {

    http:Client cli = check new http:Client("https://api.github.com");
    string path = string `/repos/${org}/${repo}/labels`;
    var process = function(json result) returns error|boolean {
                      return false;
                  };
    check invokeAPI(cli, path, process);
}

function invokeAPI(http:Client cli, string path, function (json) returns boolean|error logic) returns error? {
    json result = <json>check cli->get(path, targetType = json);
}
```

test-pack-1613805262889

Error 

```
[Error - 12:44:24 PM] Operation 'text/completion' failed! {uri: '/d%3A/test/temp.bal', [21:44], error: 'No value present'}
java.util.NoSuchElementException: No value present
	at java.base/java.util.Optional.get(Optional.java:148)
	at org.ballerinalang.langserver.completions.builder.FunctionCompletionItemBuilder.getDocumentation(FunctionCompletionItemBuilder.java:188)
	at org.ballerinalang.langserver.completions.builder.FunctionCompletionItemBuilder.setMeta(FunctionCompletionItemBuilder.java:148)
	at org.ballerinalang.langserver.completions.builder.FunctionCompletionItemBuilder.build(FunctionCompletionItemBuilder.java:90)
	at org.ballerinalang.langserver.completions.providers.AbstractCompletionProvider.populateBallerinaFunctionCompletionItem(AbstractCompletionProvider.java:328)
	at org.ballerinalang.langserver.completions.providers.AbstractCompletionProvider.lambda$getCompletionItemList$0(AbstractCompletionProvider.java:148)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1541)
	at org.ballerinalang.langserver.completions.providers.AbstractCompletionProvider.getCompletionItemList(AbstractCompletionProvider.java:143)
	at org.ballerinalang.langserver.completions.providers.AbstractCompletionProvider.expressionCompletions(AbstractCompletionProvider.java:380)
	at org.ballerinalang.langserver.completions.providers.context.CheckExpressionNodeContext.getCompletions(CheckExpressionNodeContext.java:57)
	at org.ballerinalang.langserver.completions.providers.context.CheckExpressionNodeContext.getCompletions(CheckExpressionNodeContext.java:37)
	at org.ballerinalang.langserver.completions.util.CompletionUtil.route(CompletionUtil.java:99)
	at org.ballerinalang.langserver.completions.util.CompletionUtil.getCompletionItems(CompletionUtil.java:56)
	at org.ballerinalang.langserver.completions.BallerinaCompletionExtension.execute(BallerinaCompletionExtension.java:51)
	at org.ballerinalang.langserver.completions.BallerinaCompletionExtension.execute(BallerinaCompletionExtension.java:37)
	at org.ballerinalang.langserver.LangExtensionDelegator.completion(LangExtensionDelegator.java:92)
	at org.ballerinalang.langserver.BallerinaTextDocumentService.lambda$completion$0(BallerinaTextDocumentService.java:139)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1700)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.exec(CompletableFuture.java:1692)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183)
```

### PR DESCRIPTION
**Pull request by nadeeshaan**
## Purpose
With this,
- Add chain completion support for annotation access expression
- Fix completion issues for the variable where the type is of function type descriptor
Fixes #20202 
Fixes #28788

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [x] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: language-server/modules/langserver-core/src/test/resources/completion/new/expression_context/source/field_access_ctx_source13.bal [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+public type T1 record {
+    string name;
+};
+
+T1 a = { name: "John" };
+
+function testAnnotationAccessExpression() {
+	typedesc<T1> t = typeof a;
+    int annotationVal = t.@v1.
+}
```
Comment (IMS94) [inline language-server/modules/langserver-core/src/test/resources/completion/new/expression_context/source/field_access_ctx_source13.bal:18]: Seems like this is missing a new line at the end.

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/builder/FieldCompletionItemBuilder.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+     *
+     * @param symbol record field symbol or the object field symbol
+     * @return {@link CompletionItem} generated completion item
+     */
+    private static CompletionItem getCompletionItem(Symbol symbol) {
+        String recordFieldName = symbol.getName().orElseThrow();
+
+        CompletionItem completionItem = new CompletionItem();
+        completionItem.setLabel(recordFieldName);
+        completionItem.setInsertText(recordFieldName);
```
Comment (IMS94) [inline language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/builder/FieldCompletionItemBuilder.java:45]: Can we set the the `completionItem.kind()` as well here? For record fields, it's `Struct` and for object fields it should be `Interface` We have used the same kinds at `org.ballerinalang.langserver.completions.builder.TypeCompletionItemBuilder#setMeta()`

**Review by IMS94 — APPROVED** [review]
Approved with minor changes.

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/builder/FieldCompletionItemBuilder.java (hunk 2 above)
Comment (nadeeshaan) [reply language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/builder/FieldCompletionItemBuilder.java:46]: @IMS94 thank you, I think we have missed setting the kind from the beginning. One correction here though, we have to set the kind as `Field`. We have used `Struct` for record variables and `Interface` for the Object variables. Hence I'll be setting the kind as `Field` here. Addressed with  607ade185fe7a8128e80b1a9693de3a175d5face

**Review by dulajdilshan — APPROVED** [review]

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/builder/FieldCompletionItemBuilder.java (hunk 2 above)
Comment (IMS94) [reply language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/builder/FieldCompletionItemBuilder.java:46]: @nadeeshaan I see. Great!

**Pull request merged**